labgrid/config.py

```diff
@@ -153,6 +153,64 @@ def set_option(self, name, value):
         assert isinstance(value, str)
         self.data.setdefault('options', {})[name] = value
 
+    def get_target_option(self, target, name, default=None):
+        """Retrieve an entry from the options subkey under the specified target
+           subkey
+
+        Args:
+            target (str): name of the target
+            name (str): name of the option
+            default (str): A default parameter in case the option can not be
+                found
+
+        Returns:
+            str: value of the option or default parameter
+
+        Raises:
+            KeyError: if the requested key can not be found in the
+                configuration, or if the target can not be found in the
+                configuration.
+        """
+        if target not in self.data['targets']:
+            raise KeyError("No target '{}' found in configuration".format(target))
+
+        try:
+            return str(self.data['targets'][target]['options'][name])
```

Comment:
Is there a reason why options are stringified?

Reply:
No, but the global options `get_option()` are stringified so I figured it was best to follow that.

Reply:
Fair, I wonder why we introduced this restriction in the first place.

Reply:
So i looked into this and we probably did this to ensure a consistent data type going through the config. Retrieving options from the YAML will always give you a string, since we have no way to annotate the type. Callers can easily convert the options back and raise the appropriate error message if the conversion fails. I think we should stay with str for target options as well.